modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -1367,6 +1368,35 @@ iocshEnvClear(const char *name)
     }
 }
 
+void epicsStdCall
+iocshSet(const char *name, const char * value)
+{
+    iocshContext *context;
+
+    if (iocshContextId) {
+        context = (iocshContext *) epicsThreadPrivateGet(iocshContextId);
+
+        if (context != NULL) {
+            macPutValue(context->handle, name, value);
+        }
+    }
+}
+
+void epicsStdCall
+iocshShow()
+{
+    iocshContext *context;
+
+    if (iocshContextId) {
+        context = (iocshContext *) epicsThreadPrivateGet(iocshContextId);
+
+        if (context != NULL) {
+            macReportMacros(context->handle); // returns long, do we need it?
+            // takes macReportMacros((MAC_HANDLE *handle ), where to take it from?
```

Comment:
We're going to need to do a bit more for this routine, since the output from `macReportMacros()` is more complex and shows internal information that would only confuse users. That may mean modifying macLib as well, but we can talk about that later.

Reply:
> ... shows internal information that would only confuse users ...

I won't disagree, but does macLib currently provide any other report/dump/show functions?

Can we avoid scope creep into `macCore.c`?

Reply:
@mdavidsaver Unfortunately no, and looking at the output from `macReportMacros()` using @oksanagit's now-fixed branch we definitely need new code to generate something more like the output from `var` and `epicsEnvShow` instead of this:
```
Cox$ bin/darwin-aarch64/softIoc st.cmd 
# <<< st.cmd ---
echo "IOCSH_STARTUP_SCRIPT = st.cmd"
IOCSH_STARTUP_SCRIPT = st.cmd
show
e name             rawval           value
- ----             ------           -----
s ----             ------           -----
  IOCSH_STARTUP_SCRIPT st.cmd           st.cmd
#
iocshLoad l0.cmd "level=0,TOP_STARTUP_SCRIPT=st.cmd"
# <<< l0.cmd ---
show
e name             rawval           value
- ----             ------           -----
s ----             ------           -----
  IOCSH_STARTUP_SCRIPT st.cmd           st.cmd
s ----             ------           -----
  level            0                0
  TOP_STARTUP_SCRIPT st.cmd           st.cmd
  IOCSH_STARTUP_SCRIPT l0.cmd           l0.cmd
#
iocshLoad l1.cmd "level=1,PARENT=l0.cmd"
# <<< l1.cmd ---
echo "IOCSH_STARTUP_SCRIPT=l1.cmd, level=1"
IOCSH_STARTUP_SCRIPT=l1.cmd, level=1
show
e name             rawval           value
- ----             ------           -----
s ----             ------           -----
  IOCSH_STARTUP_SCRIPT st.cmd           st.cmd
s ----             ------           -----
  level            0                0
  TOP_STARTUP_SCRIPT st.cmd           st.cmd
  IOCSH_STARTUP_SCRIPT l0.cmd           l0.cmd
s ----             ------           -----
  level            1                1
  PARENT           l0.cmd           l0.cmd
  IOCSH_STARTUP_SCRIPT l1.cmd           l1.cmd
# --- l1.cmd >>>
#
show
e name             rawval           value
- ----             ------           -----
s ----             ------           -----
  IOCSH_STARTUP_SCRIPT st.cmd           st.cmd
s ----             ------           -----
  level            0                0
  TOP_STARTUP_SCRIPT st.cmd           st.cmd
  IOCSH_STARTUP_SCRIPT l0.cmd           l0.cmd
# --- l0.cmd >>>
#
echo "IOCSH_STARTUP_SCRIPT = st.cmd"
IOCSH_STARTUP_SCRIPT = st.cmd
show
e name             rawval           value
- ----             ------           -----
s ----             ------           -----
  IOCSH_STARTUP_SCRIPT st.cmd           st.cmd
# --- st.cmd >>>
epics> exit
Cox$ cat st.cmd l0.cmd l1.cmd 
# <<< st.cmd ---
echo "IOCSH_STARTUP_SCRIPT = $(IOCSH_STARTUP_SCRIPT)"
show
#
iocshLoad l0.cmd "level=0,TOP_STARTUP_SCRIPT=$(IOCSH_STARTUP_SCRIPT)"
#
echo "IOCSH_STARTUP_SCRIPT = $(IOCSH_STARTUP_SCRIPT)"
show
# --- st.cmd >>>
# <<< l0.cmd ---
show
#
iocshLoad l1.cmd "level=1,PARENT=$(IOCSH_STARTUP_SCRIPT)"
#
show
# --- l0.cmd >>>
# <<< l1.cmd ---
echo "IOCSH_STARTUP_SCRIPT=$(IOCSH_STARTUP_SCRIPT), level=$(level)"
show
# --- l1.cmd >>>
```
The separator lines labelled `s` in the show output LH column are where each new scope begins. Look at the `show` output from the l1.cmd file; that confuses even me at times.